Include pybids as a BIDS parser

gab-berest opened the pull request:
PR that changes loading dicom to pybids package form to facilitate file parsing. Related to [https://github.com/shimming-toolbox/shimming-toolbox-py/issues/75](url).

kousu commented:
Aha: https://github.community/t5/GitHub-Actions/Self-hosted-runner-security-with-public-repositories/m-p/50172#M7723

We can use `tristano` with Github actions, but we have to make sure to *not* run them on PRs (the same way I [disabled `travis-ci/pr`](https://github.com/neuropoly/spinalcordtoolbox/issues/2674). PRs could run Octave on Github's infra, but `tristano` + matlab should be reserve to branches in the repo.

This does seem kind of fragile. It would be easy to open a cybersecurity risk since it fails-open instead of fails-closed.

jcohenadad commented:
@kousu that sounds like a reasonable approach. We can add a big warning in the "contribution" and/or wiki to explain why github actions should be disabled for PRs (until this issue is resolved-- if it is one day). So, does it mean that we won't need to rely on Azure pipeline anymore? I like the idea of not having to learn another CI environment. It also means that we can choose a more performant station (although this bit could be reserved for later, if we find tristano is too slow).

agahkarakuzu commented on azure-pipelines.yml:
```diff
@@ -0,0 +1,18 @@
+steps:
+- script: |
+    # Beware: you MUST put `quit` at the end; Matlab does not auto-quit.
+    # And worse, our Matlab license only allows one process at a time (TODO: verify this),
```
With the license we have on Tristano, concurrent runs are possible, the issue is if Azure build pipeline allows such multi-threading, as collecting unit test artifacts will need to be parallelized. If possible, we should better set`maxParallel: 2` with this machine. 

[Here](https://docs.microsoft.com/en-us/azure/devops/pipelines/licensing/concurrent-jobs?view=azure-devops#self-hosted-cicd) you can find details on parallelism.

agahkarakuzu commented on azure-pipelines.yml:
```diff
@@ -0,0 +1,18 @@
+
+# use the self-hosted runner at tristano.neuro.polymtl.ca
+# this has Matlab installed and a license to use it.
+pool: 'Default'
```
If you have non-matlab jobs, you can use a Microsoft hosted agent in parallel (you can use up to 10) to remove some burden off the shoulders of little tristano :)

agahkarakuzu commented:
🎊 badge for this branch:
[![Build Status](https://dev.azure.com/neuropoly/RTShimming/_apis/build/status/shimming-toolbox.shimming-toolbox?branchName=azure-pipelines)](https://dev.azure.com/neuropoly/RTShimming/_build/latest?definitionId=10&branchName=azure-pipelines)

Badge for master (should look nice on README):

[![Build Status](https://dev.azure.com/neuropoly/RTShimming/_apis/build/status/shimming-toolbox.shimming-toolbox?branchName=master)](https://dev.azure.com/neuropoly/RTShimming/_build/latest?definitionId=10&branchName=master)

agahkarakuzu commented:
> We can use tristano with Github Actions

@kousu @jcohenadad  FYI, GitHub actions  self-hosted runners do not have a good solution at an organization level yet. 

On Azure, you can associate certain actions with approved users, that big warning on GitHub self hosted runners + [WIP] for organization repos were repelled me from giving them a try yet.

agahkarakuzu commented:
You should definitely check this checkbox:

![image](https://user-images.githubusercontent.com/9632322/82386683-005e9780-9a03-11ea-96ac-84126df61066.png)

On each PR, someone from the team can check files changed and confirm. I turned off fork builds for good measure anyway.

jcohenadad commented:
@kousu i've extended the scope of this PR to also include doc build, hope that's ok

kousu commented:
I think the doc build part should be isolated to https://github.com/shimming-toolbox/shimming-toolbox.org/pull/31

jcohenadad committed:
Initial commit

mathieuboudreau committed:
Add basic python package setup

mathieuboudreau committed:
Create empty class

mathieuboudreau committed:
Add default zeros volume

mathieuboudreau committed:
Add basic shepp-logan init & test

mathieuboudreau committed:
Add empty line to force test Travis build

mathieuboudreau committed:
Add build badge

mathieuboudreau committed:
Merge pull request #4 from shimming-toolbox/mb/setup

Add basic python package setup

mathieuboudreau committed:
Add customized phantom and tests

mathieuboudreau committed:
Add method to generate deltaB0 & tests

mathieuboudreau committed:
Add simulation and get methods & tests

mathieuboudreau committed:
Add SNR functionality & test

mathieuboudreau committed:
Add sanity check for simulation - B0 map measurement must match generated one

mathieuboudreau committed:
Add test for right-handedness

mathieuboudreau committed:
Add save functionality + test

mathieuboudreau committed:
Add more tests

mathieuboudreau committed:
Add general signal tests

mathieuboudreau committed:
Ran Black code auto-formatter

mathieuboudreau committed:
Document class & adhere names to convention

mathieuboudreau committed:
Fix typo

kousu committed:
Sketch for some command line tools in the toolbox.

These are based on the flowchart at https://www.shimming-toolbox.org/1_overview/2_flowchart/
 (NB: that link will probably rot; the chart isn't screenshotable by archive.org either)
but it's likely that they will be deleted and replaced by scripts that perform
a larger block of the MRI shimming process in one step -- partly by calling
the (currently empty) library functions also included here.

kousu committed:
Hide .pyc's from git too

kousu committed:
Install testing dependencies as "extras".

This prevents an unnecessary weight installed by users who
probably don't want to run our unit tests; though it allows
them to, if they read the (as yet unwritten) dev docs or get
a tip from a developer helping them debug their install:
just `pip install shimmingtoolbox[testing]`.

This is the blessed method from the python packaging developers:
https://github.com/pypa/setuptools/issues/1684#issuecomment-648441338

The other(?) blessed method is to use tox's `deps` option:
https://github.com/pypa/setuptools/issues/931

kousu committed:
Bump python version backwards

Ubuntu Xenial still uses 3.6 and it's still alive for at least
half a year more.

po09i committed:
BUG : Typo

po09i committed:
BUG : Typo

gaspardcereza committed:
added first scenario (all commented)

kousu committed:
Document setup

gaspardcereza committed:
started to convert nifti_read and dicom_to_nifti into python

gaspardcereza committed:
deleted rescaling steps in read_nii because nibabel does it automatically

gaspardcereza committed:
added dataset downloading from OSF

jcohenadad committed:
Replaced OSF with github link

jcohenadad committed:
Sorting out data download and path naming

jcohenadad committed:
Update volume_RT_shim_respiratory.py

jcohenadad committed:
Added docstrings, minor comments

gaspardcereza committed:
Update volume_RT_shim_respiratory.py

Identified the function calls and added context

gaspardcereza committed:
Update volume_RT_shim_respiratory.py

Converted variable names to `snake_case`

gaspardcereza committed:
Added clarifications about the nifti_path variable

gaspardcereza committed:
Update volume_RT_shim_respiratory.py

gaspardcereza committed:
Changed some variable names, completed dicom_to_nifti

gaspardcereza committed:
updated variable names and error handling

gaspardcereza committed:
Prevent git checking in sample dataset

gaspardcereza committed:
Uptaded functions according to reviews

gaspardcereza committed:
modified config.json files generation

gaspardcereza committed:
Added possibility to not read the images when calling read_nii

jcohenadad committed:
Added test (unfinished)

kousu committed:
Source package version from setup.py. (#19)

This avoids a tricky chicken-or-egg problem:
as soon as shimmingtoolbox/__init__.py starts loading up
submodules, which are going to load up libraries,
installation will fail because setup.py was doing

> import shimmingtoolbox

at the top, just to get at

> shimmingtoolbox.__version__

and of course when setup.py is run, none of the
libraries we need will be installed yet.

This is mentioned on https://packaging.python.org/guides/single-sourcing-package-version/

> Warning
>
> Although this technique is common, beware that it will fail if sample/__init__.py
> imports packages from install_requires dependencies, which will very likely not
> be installed yet when setup.py is run.

and after looking through that page, I decided that using importlib.metadata
is the future-proofed option.

(Editorializing: I really wish PyPA would stick to the Zen of Python's admonishment
to "There should be one obvious way to do it."; but the first thing this page says is
"There are many techniques" and the best is buried as #5)

Co-authored-by: Julien Cohen-Adad <jcohen@polymtl.ca>

jcohenadad committed:
Added cli entry point for download data

jcohenadad committed:
changed cli name for st_*

jcohenadad committed:
Added cli, deps

jcohenadad committed:
Added API to deal with downloads

jcohenadad committed:
Added utils.py, collection of useful APIs

jcohenadad committed:
Added CLI for download data

jcohenadad committed:
Small fixes on output folder and temp folder name

jcohenadad committed:
Added TODOs

jcohenadad committed:
Forgot a comma :-)

jcohenadad committed:
Added download testing data in Travis

jcohenadad committed:
Removed one "script" instance

jcohenadad committed:
added click as dep

jcohenadad committed:
Added testing_data to .gitignore

jcohenadad committed:
Forgot comma

jcohenadad committed:
Added __dir_testing__ to be called by unit tests

jcohenadad committed:
Implemented unit test for dicom_to_nifti()

jcohenadad committed:
Removed examples/script_nifti.py

Now replaced by tests/test_dicom_to_nifti

jcohenadad committed:
Changed variable name

jcohenadad committed:
Added dcm2bids as a dependency

kousu committed:
Install dcm2niix in CI

kousu committed:
Avoid double-building CI on PRs

This integrates @Drulex's idea from https://github.com/neuropoly/spinalcordtoolbox/issues/2674#issuecomment-659622200

kousu committed:
Remove dead comment

gaspardcereza committed:
deleted rescaling function (now included in read_nii)

gaspardcereza committed:
removed .gitignore in /examples (testing data no more uploaded there)

gaspardcereza committed:
updated read_nii

gaspardcereza committed:
Modified dicom_to_bids to remove .json in nifti path + updated unit test

gaspardcereza committed:
corrected errors in download functions

gaspardcereza committed:
added tests to check that json files are created

gaspardcereza committed:
added test for version of dcm2bids

jcohenadad committed:
Removed test of dcm2bids version; now specified in setup

jcohenadad committed:
Refactored dicom_to_nifti

Direct use of dcm2bids with a config .json file

jcohenadad committed:
Added dcm2bids.json config file

jcohenadad committed:
Updated config file to output BIDS-compatible fieldmap

jcohenadad committed:
Working unit test

jcohenadad committed:
Cleanup

jcohenadad committed:
Removed function not used here

Also see: https://github.com/shimming-toolbox/shimming-toolbox-py/issues/33

gaspardcereza committed:
Create CONTRIBUTING.rst

Based on what has been done for SCT

gaspardcereza committed:
Added versioning + corrected minor errors

gaspardcereza committed:
Modified link to category labels

Co-authored-by: Julien Cohen-Adad <jcohen@polymtl.ca>

gaspardcereza committed:
removed function sub-labels

Co-authored-by: Julien Cohen-Adad <jcohen@polymtl.ca>

gaspardcereza committed:
added LICENSE file

This is a copy of the license file we had for the Matlab version of the toolbox

gaspardcereza committed:
added link to license file

gaspardcereza committed:
deleted files

kousu committed:
Upgrade dcm2niix to latest.

dcm2bids has only ever been tested with the latest version,
and dcm2niix is still shaking out its API bugs, so the only
reliable thing to do is to demand everyone stay up to date.

This is okay almost everywhere except on Debian which is always
slow to upgrade stuff.

kousu committed:
Put my versioning snippet back together.

jidicula committed:
Port shimming-toolbox documentation (#41)

The RTD .md files are copied over from shimming-toolbox and converted using pandoc into .rst files. RTD build steps are adapted from AxonDeepSeg. API documentation is also added.

kousu committed:
Upgrade dcm2niix to latest. (#43)

dcm2bids has only ever been tested with the latest version,
and dcm2niix is still shaking out its API bugs, so the only
reliable thing to do is to demand everyone stay up to date.

This is okay almost everywhere except on Debian which is always
slow to upgrade stuff.

Co-authored-by: Julien Cohen-Adad <jcohen@polymtl.ca>

po09i committed:
Add unwrap phase functionality (#15)

* NEW : Added unwrap folder structure

* Added numpy requirement

* Remove version from __init__ subfolder

* Initial function for unwrap phase

* NEW : Initial prelude wrapper

* NEW : fill init

* update requirements

* NEW : add debug script

* make script callable from CL

* Fix delete of files

* Added .idea

* Added download of example data

* Added phase loading and diff

* Update script to use test data

* Return 3d instead of 2d

* Added support for 5d complex arrays

* Added better error handling

* Added 2d support

* Updated according to comments

* Update todo for phaseunwrapping

* Added docstring

* moved mask to be defined in script

* moved mask to script

* added docstring

* added todo

* Update docstring

* Default to no mask

* Changed download function

* Removed unnecessary comment

* Resolve issues in script according to comments

* Update according to comments

* Removed interp function

* Added simple test

* Added deps

* clustered phase code together

* Minor fixes

* Commented out the whole function (not useable anymore)

* Rewrote prelude to have mag and phase inputs

* Removed complex array handling

* Wrote test to accomodate new prelude inputs

* Added download in travis

* forgot a comma

* and another one

* Removed reference to unwrap_phase

* set up tests with the new download function

* Added handling for non default paths

* Added test for non default path

* Added mask tests

* Added tests for mag and phase input

* Added more checks for non_default_masks

* Refactored test to remove duplication

* Now possible to download non-compressed data, e.g. prelude

* Download prelude binary

Related to https://github.com/shimming-toolbox/shimming-toolbox-py/issues/23

* Added click as a dependency

* Fixed hardcoded name when deleting unwrapped nii file

* Calls prelude in tmp folder for every test

* Added default back

* Test prelude -h

* Changed test to .yml file

* Added prelude to the path to be called within script

* Add permissions

* extra char

* Add FSL required variable

* Reduced permissions

* Clean up

* Use download function

* Updated default for mask input argument

* Added API: add_suffix()

* Refactored output of prelude() API

* Removed phase difference todo and opened issue #36

* Updates variables to reflect convention

* Removed error handling from example script

* Resolved comments

* Added 2D as an input type and changed is_unwrapping_in_2d to false

* Removed logging

* Updated tests

* Removed B0 mapping code: out of scope

* Add threshold for more precise masking

* forgot shape

* Apply suggestions from code review

Co-authored-by: Julien Cohen-Adad <jcohen@polymtl.ca>

* Added run_subprocess function

* Use run_subprocess() function

* Replaced glob.glob by __dir_testing__

* Changed matplotlib for oriented-object use

So it doesn't crash on headless systems, see: https://github.com/neuropoly/spinalcordtoolbox/wiki/Moving-away-from-pyplot

* Cosmetic fixes

* Removed unused file

This file is now referenced in #45

Co-authored-by: Julien Cohen-Adad <jcohen@polymtl.ca>

jidicula committed:
[docs][README] Add RTD build sticker

Resolves: #48

jidicula committed:
[docs][API reference] Replace Autodoc with Napoleon (#57)

**Why this change was necessary**
Sphinx's default Autodoc does not recognize Google style docstrings
and throws warnings when autogenerating API documentation.

**What this change does**
Replaces Sphinx's default Autodoc extension with Napoleon.

**Additional context/notes/links**

Resolves: #55
Related to: #53 (partially resolves), #54 (partially resolves)

See also:
https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html
https://github.com/shimming-toolbox/shimming-toolbox-py/pull/56#issuecomment-667320630

jidicula committed:
[docs][API reference] Add unwrap module to API reference (#56)

* [docs][API reference] Add unwrap module

Resolves: #52

* [docs][API reference] Add unwrap heading

**Why this change was necessary**
`unwrap.py` has a header docstring too, that renders above the
function documentation when built with Sphinx.

**What this change does**
Adds a heading for `unwrap` to distinguish it from the module above in
the documentation.

**Additional context/notes/links**

Resolves: #52

jcohenadad committed:
Added masking.threshold() API and associated test

jcohenadad committed:
Renamed to test_masking.py

So we will put all the masking algo inside this file, which makes sense because the testing data (np.array) will be used as input for all the algos.

jidicula committed:
[docs][fix] Add documentation dependency installation steps

**Why this change was necessary**
The develop installation instructions did not include the
documentation dependencies.

**What this change does**
Adds `docs` to the `pip` installation instructions.

**Any side-effects?**
None

**Additional context/notes/links**

Resolves: #61

jcohenadad committed:
Updated general demo script: general_demo.py (#46)

* Renamed to general_demo.py

This script will include most functionalities

* Added dicom_to_nifti

* Added colorbar to plots

* Removed unused dependency

* Changed CLI name to reflect name of file change

* Add a test file to test if scripts run appropriatly

* call script with function instead of CLI

* Added interp function based on comments

* Moved interp outside of function

* Added interp in prelude

* Debug travis error

* Added output path argument

* Forgot return variable name

* Refactored test

* Added default value to not delete filesin output directory

* Tidy up

* Renamed main()

* Update test to new name

* Remove fake CLI

* Added documentation for website

* Added fstring to script

* Added dependencies and example folder to autodoc

* Added autodoc and removed TODO

* Added __init__épy file to examples

* Added docs/build/ to .gitignore for local builds of website

Co-authored-by: Alexandre D'Astous <po09i@hotmail.com>

po09i committed:
Initial shape masking implementation and related test

po09i committed:
Added suggestions and new implementation

po09i committed:
Added cube mask

po09i committed:
Added related tests

po09i committed:
Fixed dimensions and added docstring

po09i committed:
Refactored tests

po09i committed:
Fixed example and added to documentation

po09i committed:
Renamed to shapes to avoid conflict with shape argument

jidicula committed:
[style][download] Reformat file with Black

jidicula committed:
[docs][download] Fix docstring formatting in `install_data()`

**Why this change was necessary**
The docstring was previously formatted incorrectly and did not comply
with Google docstring style.

**What this change does**
Refactors the docstring according to Google docstring style.

**Any side-effects?**
Sphinx warnings for the module no longer appear when building the
documentation.

**Additional context/notes/links**

Resolves: #54

jidicula committed:
[docs][download] Fix docstring formatting in `download_data()`

**Why this change was necessary**
The docstring was previously formatted incorrectly and did not comply
with Google docstring style.

**What this change does**
Refactors the docstring according to Google docstring style.

**Any side-effects?**
None

**Additional context/notes/links**

Resolves: #54

jidicula committed:
[docs][download] Fix docstring formatting in `unzip()`

**Why this change was necessary**
The docstring was previously formatted incorrectly and did not comply
with Google docstring style.

**What this change does**
Refactors the docstring according to Google docstring style.

**Any side-effects?**
None

**Additional context/notes/links**

Resolves: #54

jidicula committed:
[style][utils] Reformat with Black

jidicula committed:
[docs][utils] Fix docstring formatting in `add_suffix()`

**Why this change was necessary**
The docstring was previously formatted incorrectly and did not comply
with Google docstring style.

**What this change does**
Refactors the docstring according to Google docstring style.

**Any side-effects?**
None

**Additional context/notes/links**

Resolves: #67

jidicula committed:
[docs][utils] Fix docstring in `run_subprocess()`

**Why this change was necessary**
The docstring was previously formatted incorrectly and did not comply
with Google docstring style.

**What this change does**
Refactors the docstring according to Google docstring style.

**Any side-effects?**
None

**Additional context/notes/links**

Resolves: #67

jidicula committed:
[docs][utils] Fix docstring in `st_progress_bar()`

**Why this change was necessary**
The docstring was previously formatted incorrectly and did not comply
with Google docstring style.

**What this change does**
Refactors the docstring according to Google docstring style.

**Any side-effects?**
None

**Additional context/notes/links**

Resolves: #67

jidicula committed:
[style][b0map] Format with Black

jidicula committed:
[docs][b0map] Fix docstring in `main()`

**Why this change was necessary**
The docstring was previously formatted incorrectly and did not comply
with Google docstring style.

**What this change does**
Refactors the docstring according to Google docstring style.

**Any side-effects?**
None

**Additional context/notes/links**

Resolves: #68

jidicula committed:
[refactor][b0map] Refactor variable name

**Why this change was necessary**
`map` is a reserved keyword in Python and should not be used as a
name.

**What this change does**
Renames `map` to `fieldmap`. See diff for details.

**Any side-effects?**
None

**Additional context/notes/links**

Related to: #68

jidicula committed:
[style][download_data] Format with Black

jidicula committed:
[docs][download_data] Add CLI args to docstring

**Why this change was necessary**
The CLI help contains flags with manpage info, but those were not
replicated in the docstring and therefore would not be pulled into the
Sphinx build.

**What this change does**
Adds an `Args` section to the docstring with info about each argument.

**Any side-effects?**
None

**Additional context/notes/links**

Resolves: #69

jidicula committed:
[docs][numerical_model] Add module section title in docs

jidicula committed:
[docs][numerical_model] Correct formatting in module docstring

**Why this change was necessary**
The usage example was not rendering correctly as a code block with
syntax highlighting.

**What this change does**
Reformats the example so it renders as a code block with syntax
highlighting.

**Any side-effects?**
None.

**Additional context/notes/links**

Resolves: #53

jidicula committed:
[docs][NumericalModel] Fix formatting in autodoc class docstring

**Why this change was necessary**
The previous docstring did not comply with Google docstring formatting
and was not rendering correctly.

**What this change does**
Fixes docstring formatting to comply wth Google docstring style and
therefore render correctly when built by Sphinx.

**Any side-effects?**
None

**Additional context/notes/links**

Resolves: #53

jidicula committed:
[docs][NumericalModel] Fix formatting in `generate_deltaB0()`

**Why this change was necessary**
The previous docstring was not adhering to Google docstring style and
was therefore rendering incorrectly in the Sphinx build.

**What this change does**
Corrects formatting of the docstring so it renders correctly in the
Sphinx build.

**Any side-effects?**
None.

**Additional context/notes/links**

Resolves: #53

jidicula committed:
[docs][masking.shapes] Fix automodule stub

jidicula committed:
[feat][RTD build] Reconfigure RTD build to fail on warning

**Why this change was necessary**
Previously, the RTD build allowed warnings, meaning that badly
formatted docstrings could enter the production documentation unless a
contributor remembered to check the documentation build messages. By
configuring a stricter build check, a contributor is required to
correctly structure the accompanying documentation for any features
they add to the software.

**What this change does**
See title.

**Any side-effects?**
PRs with bad docstring formatting WILL NOT pass RTD checks from now on.

**Additional context/notes/links**

How to build docs locally:
https://github.com/shimming-toolbox/shimming-toolbox-py/wiki

jidicula committed:
Revert "[style][download_data] Format with Black"

This reverts commit 7d3cb3b9305858e83fc972680ca3e288df362029.

Addresses: https://github.com/shimming-toolbox/shimming-toolbox-py/pull/72#discussion_r470665353

jidicula committed:
Revert "[style][b0map] Format with Black"

This reverts commit 2688caf824897531f5541ac8d35593bdc39a0683.

Addresses: https://github.com/shimming-toolbox/shimming-toolbox-py/pull/72#discussion_r470665353

jidicula committed:
Revert "[style][utils] Reformat with Black"

This reverts commit de05090595251259ef4fb271b8d50c79b613cba5.

Addresses: https://github.com/shimming-toolbox/shimming-toolbox-py/pull/72#discussion_r470665353

jidicula committed:
Revert "[style][download] Reformat file with Black"

This reverts commit 21f59995e127ebb6575a6abab2f09dc39d34a389.

Addresses: https://github.com/shimming-toolbox/shimming-toolbox-py/pull/72#discussion_r470665353

jidicula committed:
[docs][API] Rename masking module heading

Addresses: https://github.com/shimming-toolbox/shimming-toolbox-py/pull/72\#discussion_r470671456

gab-berest committed:
Add the ability to load niftis (#35)

* Copy in python syntax existing function (Not working)

* Updated function header (Still not working)

* Added parsing in acquisition folders

* Added read_nii file and revert to remove merge things

* Remove CLI

* Working load_nifti and added test architecture and mock values

* Rework mock data and setup

* Finished test for load_nifti with dummy data

* Create CONTRIBUTING.rst

Based on what has been done for SCT

* Added versioning + corrected minor errors

* Modified link to category labels

Co-authored-by: Julien Cohen-Adad <jcohen@polymtl.ca>

* removed function sub-labels

Co-authored-by: Julien Cohen-Adad <jcohen@polymtl.ca>

* added LICENSE file

This is a copy of the license file we had for the Matlab version of the toolbox

* added link to license file

* Upgrade dcm2niix to latest.

dcm2bids has only ever been tested with the latest version,
and dcm2niix is still shaking out its API bugs, so the only
reliable thing to do is to demand everyone stay up to date.

This is okay almost everywhere except on Debian which is always
slow to upgrade stuff.

* started to convert nifti_read and dicom_to_nifti into python

* deleted rescaling steps in read_nii because nibabel does it automatically

* added dataset downloading from OSF

* Replaced OSF with github link

* Sorting out data download and path naming

* Added docstrings, minor comments

* Changed some variable names, completed dicom_to_nifti

* updated variable names and error handling

* Prevent git checking in sample dataset

* Uptaded functions according to reviews

* modified config.json files generation

* Added possibility to not read the images when calling read_nii

* Added download testing data in Travis

* Removed one "script" instance

* added click as dep

* Added testing_data to .gitignore

* Forgot comma

* Added __dir_testing__ to be called by unit tests

* Implemented unit test for dicom_to_nifti()

* Removed examples/script_nifti.py

Now replaced by tests/test_dicom_to_nifti

* Changed variable name

* Added dcm2bids as a dependency

* deleted rescaling function (now included in read_nii)

* removed .gitignore in /examples (testing data no more uploaded there)

* updated read_nii

* Modified dicom_to_bids to remove .json in nifti path + updated unit test

* corrected errors in download functions

* added tests to check that json files are created

* added test for version of dcm2bids

* Removed test of dcm2bids version; now specified in setup

* Refactored dicom_to_nifti

Direct use of dcm2bids with a config .json file

* Added dcm2bids.json config file

* Updated config file to output BIDS-compatible fieldmap

* Working unit test

* Cleanup

* deleted files

* Put my versioning snippet back together.

* Port shimming-toolbox documentation (#41)

The RTD .md files are copied over from shimming-toolbox and converted using pandoc into .rst files. RTD build steps are adapted from AxonDeepSeg. API documentation is also added.

* Upgrade dcm2niix to latest. (#43)

dcm2bids has only ever been tested with the latest version,
and dcm2niix is still shaking out its API bugs, so the only
reliable thing to do is to demand everyone stay up to date.

This is okay almost everywhere except on Debian which is always
slow to upgrade stuff.

Co-authored-by: Julien Cohen-Adad <jcohen@polymtl.ca>

* Add unwrap phase functionality (#15)

* NEW : Added unwrap folder structure

* Added numpy requirement

* Remove version from __init__ subfolder

* Initial function for unwrap phase

* NEW : Initial prelude wrapper

* NEW : fill init

* update requirements

* NEW : add debug script

* make script callable from CL

* Fix delete of files

* Added .idea

* Added download of example data

* Added phase loading and diff

* Update script to use test data

* Return 3d instead of 2d

* Added support for 5d complex arrays

* Added better error handling

* Added 2d support

* Updated according to comments

* Update todo for phaseunwrapping

* Added docstring

* moved mask to be defined in script

* moved mask to script

* added docstring

* added todo

* Update docstring

* Default to no mask

* Changed download function

* Removed unnecessary comment

* Resolve issues in script according to comments

* Update according to comments

* Removed interp function

* Added simple test

* Added deps

* clustered phase code together

* Minor fixes

* Commented out the whole function (not useable anymore)

* Rewrote prelude to have mag and phase inputs

* Removed complex array handling

* Wrote test to accomodate new prelude inputs

* Added download in travis

* forgot a comma

* and another one

* Removed reference to unwrap_phase

* set up tests with the new download function

* Added handling for non default paths

* Added test for non default path

* Added mask tests

* Added tests for mag and phase input

* Added more checks for non_default_masks

* Refactored test to remove duplication

* Now possible to download non-compressed data, e.g. prelude

* Download prelude binary

Related to https://github.com/shimming-toolbox/shimming-toolbox-py/issues/23

* Added click as a dependency

* Fixed hardcoded name when deleting unwrapped nii file

* Calls prelude in tmp folder for every test

* Added default back

* Test prelude -h

* Changed test to .yml file

* Added prelude to the path to be called within script

* Add permissions

* extra char

* Add FSL required variable

* Reduced permissions

* Clean up

* Use download function

* Updated default for mask input argument

* Added API: add_suffix()

* Refactored output of prelude() API

* Removed phase difference todo and opened issue #36

* Updates variables to reflect convention

* Removed error handling from example script

* Resolved comments

* Added 2D as an input type and changed is_unwrapping_in_2d to false

* Removed logging

* Updated tests

* Removed B0 mapping code: out of scope

* Add threshold for more precise masking

* forgot shape

* Apply suggestions from code review

Co-authored-by: Julien Cohen-Adad <jcohen@polymtl.ca>

* Added run_subprocess function

* Use run_subprocess() function

* Replaced glob.glob by __dir_testing__

* Changed matplotlib for oriented-object use

So it doesn't crash on headless systems, see: https://github.com/neuropoly/spinalcordtoolbox/wiki/Moving-away-from-pyplot

* Cosmetic fixes

* Removed unused file

This file is now referenced in #45

Co-authored-by: Julien Cohen-Adad <jcohen@polymtl.ca>

* [docs][README] Add RTD build sticker

Resolves: #48

* [docs][API reference] Replace Autodoc with Napoleon (#57)

**Why this change was necessary**
Sphinx's default Autodoc does not recognize Google style docstrings
and throws warnings when autogenerating API documentation.

**What this change does**
Replaces Sphinx's default Autodoc extension with Napoleon.

**Additional context/notes/links**

Resolves: #55
Related to: #53 (partially resolves), #54 (partially resolves)

See also:
https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html
https://github.com/shimming-toolbox/shimming-toolbox-py/pull/56#issuecomment-667320630

* [docs][API reference] Add unwrap module to API reference (#56)

* [docs][API reference] Add unwrap module

Resolves: #52

* [docs][API reference] Add unwrap heading

**Why this change was necessary**
`unwrap.py` has a header docstring too, that renders above the
function documentation when built with Sphinx.

**What this change does**
Adds a heading for `unwrap` to distinguish it from the module above in
the documentation.

**Additional context/notes/links**

Resolves: #52

* Added the volume dimension for 5D data

* Reshape test to have a 5D ndarray

* Add test for volumes

* Correct typo

* Removed testing folder that was crashing

* Add f-string and review code

* Remove entry point form file

* Changed to separate runs

* Update test for new architecture and syntax bug

* Update tests

* Update tests for echo number

* Fix print error

* Add modality verification and correct chape error

* Fix tests for modality

* Fix syntax error

* Repair test KeyError

* Fix print issue

* Change modality_check

* Fix run test

* Fix select_run

* Fix test run

* Fix wrong path

* Fix wrong image comments

* Fix Acquisition number

* Change phase scaling condition

* Add automodule to documentation

* Rework names

* Update docstring

* Update description

* Remove EchoNumber check

* Update docstring

Co-authored-by: Julien Cohen-Adad <jcohen@polymtl.ca>

* Update docstring

Co-authored-by: Julien Cohen-Adad <jcohen@polymtl.ca>

* Changed duplicated code

Co-authored-by: Gaspard Cereza <54723104+gaspardcereza@users.noreply.github.com>
Co-authored-by: Julien Cohen-Adad <jcohen@polymtl.ca>
Co-authored-by: Nick Guenther <nguenthe@uwaterloo.ca>
Co-authored-by: Gaspard Cereza <gaspard.cereza@gmail.com>
Co-authored-by: Johanan Idicula <johanan.idicula@gmail.com>
Co-authored-by: Alex Dastous <47249340+po09i@users.noreply.github.com>

gab-berest committed:
Reproduce old code and uncomment mandatory code

gab-berest committed:
Removed config file auto generation

gab-berest committed:
Add directory check

gab-berest committed:
Remove tmp_directory if user asks it

gab-berest committed:
Update code with checks (commented code for python package use since dcm2bids.scaffold is broken as import)

gab-berest committed:
Update scaffold bug

gab-berest committed:
Update helper path name

gab-berest committed:
Change main function to pybids

gab-berest committed:
Add pybids package
